### DISCUSSION TITLE
Past "State" timeline

### DISCUSSION THREAD

**Discussion by youzer-name** [jasonacox/Powerwall-Dashboard#90] [Ideas]
I was playing around with two versions of a State Timeline panel that would show something like what is in the "Current State" panel, but for past time periods.

One option for displaying it is next to the existing Current State panel. That looks like this:

![image](https://user-images.githubusercontent.com/57046429/192814410-ece1c2e6-a37a-410c-a20c-16d82a0a036f.png)

The State Timeline is set up to show the same colors as the Current State, In this case it is showing the last 24 hours. The "Home" line is always blue so it doesn't relay much information, but I included it to match the current state panel.

The other option is to display it below the Energy Usage panel. That looks like this:

![image](https://user-images.githubusercontent.com/57046429/192818115-755e39d2-f746-47ca-bb50-c7031d60ce09.png)

The x-axis times don't exactly line up because there is no right-side scale on the state timeline, In this case I removed the Home bar because it didn't add anything to the visualization.

Would there be any interest in adding this to the main Powerwall Power Flow dashboard?

**Comment by BMWE85** [comment]
Thank you for working on this. I would be very interested in this Feature.

**Comment by youzer-name** [comment]
@BMWE85 - After I originally posted, I somehow managed to delete this panel from my dashboard. I've recreated it like the images above, except Powerwall now shows as orange when discharging.

Anything red/yellow/orange is a source of power to the house. Red is using grid, orange is discharging battery, yellow is generating solar.

Green and purple are power being exported/stored.

You have to ignore small values for grid and powerwall or it looks like they're never idle. How it is set for now
- Powerwall - from -50 (discharging) to +20 (charging) watts is 'idle'
- Grid - from -250 (generating) to +250 (using) watts is 'idle'
- Solar - 10 watts or above is 'generating'

To change any of the colors or thresholds, adjust the value mappings found in each field's overrides. Be careful adjusting the threshold values. Grafana likes to remove the leading minus sign when you try to type a negative value which can mess things up.

A dashboard json containing this single panel is attached to this post.

[Past State panel-1669480744385.json.txt](https://github.com/jasonacox/Powerwall-Dashboard/files/10096647/Past.State.panel-1669480744385.json.txt)

**Reply by BMWE85** [reply]
That's fantastic :-) Thank you very  much.
I integrated it into my Dashboard and I love it.
